osmoutils/compare.go

```diff
@@ -0,0 +1,69 @@
+package osmoutils
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+// Max returns the maximum value among the given values of any type that supports comparison.
+func Max(values ...interface{}) interface{} {
```

Comment:
What I meant by suggesting generics is something like this:

```go
type Comparable interface {
    type int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, uintptr, float32, float64, string
}

func Max[T Comparable](values []T) (T, error) {
    if len(values) == 0 {
        return nil, fmt.Errorf("empty slice provided")
    }

    max := values[0]
    for _, value := range values[1:] {
        if value > max {
            max = value
        }
    }

    return max, nil
}
```

Would that work?

Reply:
No, because the one I implemented works with both sdk values and native values, while this would only work with natives (uses > rather than GT)

Reply:
I would like to suggest instead implementing a separate generic function for a `ComparableSDK` interface that defines `GT` API.

The reason for this is that casting is a code smell around type safety concerns. It's also not as readable as a simple generic solution.

I think it's important for us to implement our utility functions with the best practice since they have a high likelihood of being used across many parts of the system

Reply:
I was not able to get your suggestion to compile. I removed the sdk values from my implementation and only used native types.

Reply:
I got this suggestion to compile with numbers:
```go
type ComparableNum interface {
	int | int8 | int16 | int32 | int64 | uint | uint8 | uint16 | uint32 | uint64 | uintptr | float32 | float64
}

func Max[T ComparableNum](values []T) (T, error) {
	if len(values) == 0 {
		return 0, fmt.Errorf("empty slice provided")
	}

	max := values[0]
	for _, value := range values[1:] {
		if value > max {
			max = value
		}
	}

	return max, nil
}

```

Realized the challenges of getting it to work for the SDK types.

Let's treat this comment as non-blocking, please. I will think more about this